sockeye/data_io.py

```diff
@@ -479,24 +533,63 @@ def _assign_to_buckets(self, source_sentences, target_sentences):
         logger.info("fill up mode: %s", self.fill_up)
         logger.info("")
 
+    def _populate_bucket_batch_sizes(self):
+        """
+        Number of sentences and words per batch for each bucket
```

Comment:
It would be great to a little bit more documentation for this function, e.g. explicitly state what member it sets (on a related note: would it be possible to make this function return bucket_batch_sizes and set it explicitly where called?) and what it does for both batching cases: (1) Sentence-based batch size: all bucket batch sizes are the same, namely set to self.batch_size. (2) ?

Reply:
The method is pretty heavily commented but I can add a few notes to the docstring as well.